wrapper/src/main/java/software/amazon/jdbc/plugin/srw/SimpleReadWriteSplittingPlugin.java

```diff
@@ -0,0 +1,678 @@
+import software.amazon.jdbc.hostavailability.HostAvailability;
+import software.amazon.jdbc.plugin.AbstractConnectionPlugin;
+import software.amazon.jdbc.plugin.AuroraInitialConnectionStrategyPlugin;
+import software.amazon.jdbc.plugin.failover.FailoverSQLException;
+import software.amazon.jdbc.plugin.readwritesplitting.ReadWriteSplittingSQLException;
+import software.amazon.jdbc.util.Messages;
+import software.amazon.jdbc.util.RdsUrlType;
+import software.amazon.jdbc.util.RdsUtils;
+import software.amazon.jdbc.util.SqlState;
+import software.amazon.jdbc.util.StringUtils;
+import software.amazon.jdbc.util.WrapperUtils;
+
+public class SimpleReadWriteSplittingPlugin extends AbstractConnectionPlugin
+    implements CanReleaseResources {
+
+  private static final Logger LOGGER = Logger.getLogger(SimpleReadWriteSplittingPlugin.class.getName());
+  private static final Set<String> subscribedMethods =
+      Collections.unmodifiableSet(new HashSet<String>() {
+        {
+          add(JdbcMethod.CONNECT.methodName);
+          add(JdbcMethod.INITHOSTPROVIDER.methodName);
+          add(JdbcMethod.NOTIFYCONNECTIONCHANGED.methodName);
+          add(JdbcMethod.CONNECTION_SETREADONLY.methodName);
+          add(JdbcMethod.CONNECTION_CLEARWARNINGS.methodName);
+          add(JdbcMethod.STATEMENT_EXECUTE.methodName);
+          add(JdbcMethod.STATEMENT_EXECUTEQUERY.methodName);
+          add(JdbcMethod.STATEMENT_EXECUTEBATCH.methodName);
+          add(JdbcMethod.STATEMENT_EXECUTEUPDATE.methodName);
+          add(JdbcMethod.PREPAREDSTATEMENT_EXECUTE.methodName);
+          add(JdbcMethod.PREPAREDSTATEMENT_EXECUTEUPDATE.methodName);
+          add(JdbcMethod.PREPAREDSTATEMENT_EXECUTELARGEUPDATE.methodName);
+          add(JdbcMethod.PREPAREDSTATEMENT_EXECUTEQUERY.methodName);
+          add(JdbcMethod.PREPAREDSTATEMENT_EXECUTEBATCH.methodName);
+          add(JdbcMethod.CALLABLESTATEMENT_EXECUTE.methodName);
+          add(JdbcMethod.CALLABLESTATEMENT_EXECUTEQUERY.methodName);
+          add(JdbcMethod.CALLABLESTATEMENT_EXECUTELARGEUPDATE.methodName);
+          add(JdbcMethod.CALLABLESTATEMENT_EXECUTEBATCH.methodName);
+          add(JdbcMethod.CALLABLESTATEMENT_EXECUTEUPDATE.methodName);
+          add(JdbcMethod.CONNECTION_SETAUTOCOMMIT.methodName);
+        }
+      });
+
+  private final PluginService pluginService;
+  private final Properties properties;
+  private final RdsUtils rdsUtils = new RdsUtils();
+  private final boolean verifyNewConnections;
+  private volatile boolean inReadWriteSplit = false;
+  private HostListProviderService hostListProviderService;
+  private Connection writerConnection;
+  private Connection readerConnection;
+  private final String writeEndpoint;
+  private final String readEndpoint;
+  private HostSpec readEndpointHostSpec;
+  private HostSpec writeEndpointHostSpec;
+  private final AuroraInitialConnectionStrategyPlugin.VerifyOpenedConnectionType verifyOpenedConnectionType;
```

Comment:
Can we move VerifyOpenedConnectionType to its own separate class since it is now shared? The VERIFY_OPENED_CONNECTION_TYPE property can be moved there as well